### ISSUE TITLE
Indent rule doesn’t validate multiline expressions

### ISSUE THREAD

**Issue by lo1tuma** [eslint/eslint#1801]
The following example has no error:

``` js
if (true) {
    a = (
1 +
2);
}
```

I would expect the following errors:

``` sh
3:0  error  Expected indentation of 8 characters  indent
4:0  error  Expected indentation of 8 characters  indent
```
##

**Comment by message** [comment]
Also think about error messages. When i have `[2, "tab"]` rule enabled, error message "Expected indentation of 1 characters" doesn't trigger any clues in my mind about tabs.

**Comment by remitbri** [comment]
Likewise, in a jsx file, in the render method, no error message when writing
```
  render() {
    return <div>
<h2>My app </h2>
  <p>{this.state.foo}</p>
          <p>{this.state.bar}</p>
    </div>
  }
```

**Comment by nzakas** [comment]
This is a known issue with the current implementation. Patches welcome!

**Comment by czjvic** [comment]
Some news? It's very painful issue.

**Comment by BYK** [comment]
I tried to work on this but it deemed quite hard. I'll try another approach but I don't think I can resolve this soon.

**Comment by nzakas** [comment]
Yeah, I'd imagine this will be open for quite some time. It's pretty difficult to do correctly, but patches always welcome.

**Comment by czjvic** [comment]
If there is some progress by BYK, will you create a branch?

**Comment by BYK** [comment]
@czjvic - nothing to share at the moment unfortunately :(

**Comment by ilyavolodin** [comment]
Here's a small repro case:

Original code:

/* eslint indent: [2, 4] */
var a = b.extend(
{
    test: function() {
        console.log("test");
    }
});
Will report the following:

  3:1  error  Expected indentation of 4 space characters but found 0  indent
  4:2  error  Expected indentation of 8 space characters but found 0  indent
  5:3  error  Expected indentation of 4 space characters but found 0  indent
  7:1  error  Expected indentation of 4 space characters but found 0  indent
First attempt at autofixing will result in no error messages and the following code:

/* eslint indent: [2, 4] */
var a = b.extend(
    {
            test: function() {
            console.log("test");
    }
    });
If you run eslint again on the generated code you will get:

  4:10  error  Expected indentation of 8 space characters but found 0  indent
  5:7   error  Expected indentation of 4 space characters but found 0  indent
Running autofix again will result in completely screwed up code:

/* eslint indent: [2, 4] */
var a = b.extend(
    {
                    test: function() {
                console.log("test");
    }
    });
Every consecutive run from there on will only make things worse.

Using ESLint v2.0.0-beta.1

**Comment by Trott** [comment]
I'm working on something to hopefully address part of this issue, specifically enforcing indentation of multi-line chained properties. So the end user will be able to choose this style:

```js
foo()
.bar
.baz();
```

Or this style instead:

```js
foo()
  .bar
  .baz();
```

And so on...

**Comment by 5minpause** [comment]
@Trott You are still working on this feature, right? Did a quick check of your fork. I hope to hear about a merged PR via this issue. Thanks for your work.
If you need some help on this somehow I'd like to help.

**Comment by Trott** [comment]
@5minpause The work I did was merged in 95ea25a45ee4ae327f2e306272d39384e5d4602f. I imagine it will be in the version of ESLint that comes out in a few days.

**Comment by 5minpause** [comment]
Woah awesome. Thanks @Trott

**Comment by kentor** [comment]
If this issue is addressed, it would report errors for this right?

```js
    if (
!this.state.desiredDesktop &&
        this.state.queryParams.specifiesDesktop() &&
        this.state.queryParams.hasResolved(props)
) {
      ...
    }
```

The above should be fixed to
```js
    if (
      !this.state.desiredDesktop &&
      this.state.queryParams.specifiesDesktop() &&
      this.state.queryParams.hasResolved(props)
    ) {
      ...
    }
```

**Comment by not-an-aardvark** [comment]
@kentor No, but that issue is also known and will hopefully be fixed relatively soon. (See https://github.com/eslint/eslint/issues/6007 and https://github.com/eslint/eslint/issues/7274.)

**Comment by benderTheCrime** [comment]
Any indication as to when we can expect the fix associated with this will be available?
